tests/sparseml/evaluation/integrations/test_perplexity.py

```diff
@@ -0,0 +1,210 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# flake8: noqa: F401
+
+
+import os
+
+import pytest
+
+from sparseml.evaluation.integrations.perplexity import perplexity_eval
+
+
+def perplexity_dependencies_available():
+    try:
+        import datasets
+        import numpy
+        import torch
+        import transformers
+    except ImportError:
+        return False
+    return True
+
+
+@pytest.mark.skipif(
+    os.getenv("NM_ML_SKIP_PYTORCH_TESTS", False),
+    reason="Skipping pytorch tests",
+)
+@pytest.mark.skipif(
+    not perplexity_dependencies_available(),
+    reason="Skipping perplexity tests due to missing dependencies",
+)
+def test_perplexity_against_huggingface():
+    target = "Xenova/llama2.c-stories15M"
+    batch_size = 4
+    nsamples = 100
+    dataset = "wikitext"
+    dataset_config_name = "wikitext-2-raw-v1"
+    split = "test"
+
+    # run through the evaluation
+    actual = round(
+        perplexity_eval(
+            target=target,
+            batch_size=batch_size,
+            nsamples=nsamples,
+            datasets=dataset,
+        ).raw["mean_perplexity"],
+        2,
+    )
+
+    # compare to the huggingface evaluation
+    expected = _huggingface_perplexity_eval(
+        dataset=dataset,
+        dataset_config_name=dataset_config_name,
+        split=split,
+        target=target,
+        batch_size=batch_size,
+        nsamples=nsamples,
+    )
+
+    assert actual == expected
+
+
+def _huggingface_perplexity_eval(
+    dataset, dataset_config_name, split, target, batch_size, nsamples
+):
+    import numpy
+    import torch
+    import tqdm
+    from datasets import load_dataset
+    from torch.nn import CrossEntropyLoss
+    from transformers import AutoModelForCausalLM, AutoTokenizer
+
+    # _compute() is taken as is from
+    # https://github.com/huggingface/evaluate/metrics/perplexity/perplexity.py
```

Comment:
let's consider adding this as a dev dependency so we will have an up to date source of truth and clean up this file a bit